CMakeLists.txt

```diff
@@ -446,12 +446,18 @@ if(NOT IREE_ENABLE_TSAN STREQUAL IREE_BYTECODE_MODULE_ENABLE_TSAN)
       "modules (controlled by IREE_BYTECODE_MODULE_ENABLE_TSAN)")
 endif()
 
-if(IREE_BYTECODE_MODULE_ENABLE_ASAN OR IREE_BYTECODE_MODULE_ENABLE_TSAN)
-  if(NOT IREE_BYTECODE_MODULE_FORCE_LLVM_SYSTEM_LINKER)
-    message(SEND_ERROR
-        "When IREE_BYTECODE_MODULE_ENABLE_ASAN or IREE_BYTECODE_MODULE_ENABLE_TSAN is ON, "
-        "IREE_BYTECODE_MODULE_FORCE_LLVM_SYSTEM_LINKER must also be ON. "
-        "ASAN/TSAN instrumentation is not currently supported in embedded modules.")
+if(NOT IREE_BUILD_COMPILER)
+  message(STATUS "Building only runtime artifacts - skipping to check if the "
+      "LLVM system linker has been enabled when IREE modules in tests are "
+      "built with ASAN/TSAN instrumentation.")
```

Comment:
I'd rather not include this sort of configuration-specific advisory log message for all builds with `-DIREE_BUILD_COMPILER=OFF`. Take this runtime-only CI build as an example: https://github.com/openxla/iree/actions/runs/6869927026/job/18683754442?pr=15588#step:4:179. The user there has explicitly configured to turn the compiler off, and this message isn't really adding much information that they aren't already aware of.

How about either converting that message into a comment:
```cmake
if(NOT IREE_BUILD_COMPILER)
  # Building only runtime artifacts - skipping to check if the
  # LLVM system linker has been enabled when IREE modules in tests are 
  # built with ASAN/TSAN instrumentation.
else()
  ...
```

or folding the branch down:
```cmake
if (IREE_BUILD_COMPILER)
  if(IREE_BYTECODE_MODULE_ENABLE_ASAN OR IREE_BYTECODE_MODULE_ENABLE_TSAN)
    if(NOT IREE_BYTECODE_MODULE_FORCE_LLVM_SYSTEM_LINKER)
      message(SEND_ERROR
          "When IREE_BYTECODE_MODULE_ENABLE_ASAN or IREE_BYTECODE_MODULE_ENABLE_TSAN is ON, "
          "IREE_BYTECODE_MODULE_FORCE_LLVM_SYSTEM_LINKER must also be ON. "
          "ASAN/TSAN instrumentation is not currently supported in embedded modules.")
    endif()
  endif()
endif()
```

Reply:
I chose the second alternative. However, I wondered about the limit on the depth of nested if blocks and readability.
There are already 3 nested if blocks in: https://github.com/openxla/iree/pull/15588/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aL912

The linter does not complain, and I think the added folded branch is readable.